torch/distributed/_tensor/_dispatch.py

```diff
@@ -309,16 +309,15 @@ def unwrap_to_op_info(
 
         for arg in args_list:
             if isinstance(arg, dtensor.DTensor):
-                args_schema.append(arg._spec)
                 local_args.append(arg._local_tensor)
-                if mesh is not None:
-                    if mesh != arg.device_mesh:
-                        raise NotImplementedError(
-                            f"{op_call}: DTensor does not support cross-mesh operation yet!"
-                            f"Got meshes: {mesh} {arg.device_mesh}"
-                        )
+                if mesh is not None and mesh != arg.device_mesh:
+                    spec = self._try_replicate_dtensor_spec_in_missing_dim(
```

Comment:
Here are we assuming the first DTensor `arg.device_mesh`'s `ndim` is the largest? What if the first is not the largest?

Reply:
Ye. This is currently a limitation, which I don't have a good solution yet. Currently, with this change, we can handle most of the cases except the case that the first DTensor's ndim is not the largest. 
[2D_DTensor, 1D_DTensor...] ---> Implicit_replication() can handle this.
[1D_DTensor, 2D_DTensor...] ---> Implicit_replication() can't handle this. 

I could potentially check if the current arg's device mesh has a root mesh, but I also don't want to replicate if the entire list is 1D DTensor, even they have a root mesh. 
e.g. [1D_DTensor, 1D_DTensor]

To handle this case [1D_DTensor, 2D_DTensor...], I would need to loop through the args to record the largest mesh first, but that would mean add an additional for-loop so I am hesitant to do that.

Reply:
sounds good to me. Can we clearly document this? This would make it easier when we revisit, e.g. to support more general cases.

Reply:
> sounds good to me. Can we clearly document this? This would make it easier when we revisit, e.g. to support more general cases.

Makes sense. Added a TODO to document the limitation.